### PR TITLE
Converted PDBBind loader

### PR DESCRIPTION
**Pull request by ncfrey**
## Description
This is a fresh PR with a new PDBBind Molnet loader using the new template and complex featurizers. The loader allows the user to download the "refined" set of 4852 protein-ligand complexes and the "general" set of 12800 complexes. The sets are described and citations are provided in the docstring.

Addresses #2279 and partially addresses #2192.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

### CODE REVIEW

**Review by rbharath — APPROVED** [review]
This is awesome! One general thought is that it might be really interesting in a future PR to add the protein-protein, protein-nucleic-acid, and nucleic-acid-ligand PDBBind sets as well to test the generality of the interaction fingerprints.

I think this is basically good to merge in, but will hold off in case other folks have any comments :)

**Comment by ncfrey** [comment]
Great! It should be easy to add those datasets in a future PR. I'm not sure if they are already in the DeepChem datasets S3 bucket? The `.tar.gz` filenames are a little opaque so it would just require some digging through those files.

**Comment by rbharath** [comment]
Ah good point! I might possibly have uploaded them already but not sure off hand. I do recall that the data layout for these other datasets is a little different than for protein-ligand (but not dramatically so) so it should hopefully not be too large a effort

File: deepchem/molnet/load_function/pdbbind_datasets.py [hunk 1]
```diff
@@ -1,458 +1,164 @@
...
-  dataset = deepchem.data.DiskDataset.from_numpy(features, labels)
-  # No transformations of data
-  transformers = []
-  if split == None:
-    return pdbbind_tasks, (dataset, None, None), transformers
+  Parameters
+  ----------
+  featurizer: Featurizer or str
+    the featurizer to use for processing the data.  Alternatively you can pass
+    one of the names from dc.molnet.featurizers as a shortcut.
```
Comment (peastman) [inline deepchem/molnet/load_function/pdbbind_datasets.py:113]: This description needs to be updated for this class.  It has to be a ComplexFeaturizer.

**Comment by peastman** [comment]
Looks good!

File: deepchem/molnet/load_function/pdbbind_datasets.py (hunk 1 above)
Comment (ncfrey) [reply deepchem/molnet/load_function/pdbbind_datasets.py:113]: Thanks for this catch, @peastman! I'll fix this and then merge it in.

**Comment by ncfrey** [comment]
It looks like there was a Travis CI error during the build. I'm going to go ahead and merge in so I can start experimenting with this loader in Colab, since the errors are unrelated to this PR.

**Pull request merged**